flask_googlemaps/__init__.py

```diff
@@ -81,6 +86,181 @@ def add_marker(self, lat=None, lng=None, **kwargs):
+                             east,
+                             stroke_color='#FF0000',
+                             stroke_opacity=.8,
+                             stroke_weight=2,
+                             fill_color='#FF0000',
+                             fill_opacity=.3,
+                             ):
+        """ Set a dictionary with the javascript class Rectangle parameters
+
+        This function sets a default drawing configuration if the user just
+        pass the rectangle bounds, but also allows to set each parameter
+        individually if the user wish so.
+
+        Args:
+            north (float): The north latitude bound
+            west (float): The west longitude bound
+            south (float): The south latitude bound
+            east (float): The east longitude bound
+            stroke_color (str): Sets the color of the rectangle border using
+                hexadecimal color notation
+            stroke_opacity (float): Sets the opacity of the rectangle border
+                in percentage. If stroke_opacity = 0, the border is transparent
+            stroke_weight (int): Sets the stroke girth in pixels.
+            fill_color (str): Sets the color of the rectangle fill using
+                hexadecimal color notation
+            fill_opacity (float): Sets the opacity of the rectangle fill
+
+
+        """
+        rectangle = {
+            'stroke_color': stroke_color,
+            'stroke_opacity': stroke_opacity,
+            'stroke_weight': stroke_weight,
+            'fill_color': fill_color,
+            'fill_opacity': fill_opacity,
+            'bounds': {'north': north,
+                       'west': west,
+                       'south': south,
+                       'east': east,
+                       }
+        }
+
+        return rectangle
+
+    def add_rectangle(self,
+                      north=None,
+                      west=None,
+                      south=None,
+                      east=None,
+                      **kwargs):
+        """ Adds a rectangle dict to the Map.rectangles attribute
+
+        The Google Maps API describes a rectangle using the LatLngBounds
+        object, which defines the bounds to be drawn. The bounds use the
+        concept of 2 delimiting points, a northwest and a southeast points,
+        were each coordinate is defined by each parameter.
+
+        It accepts a rectangle dict representation as well.
+
+        Args:
+            north (float): The north latitude
+            west (float): The west longitude
+            south (float): The south latitude
+            east (float): The east longitude
+
+        .. _LatLngBoundsLiteral:
+            https://developers.google.com/maps/documentation/javascript/reference#LatLngBoundsLiteral
+
+        .. _Rectangles:
+            https://developers.google.com/maps/documentation/javascript/shapes#rectangles
+        """
+
+        if north:
+            kwargs['bounds']['north'] = north
```

Comment:
```python
kwargs['bounds']
```
will raise KeyError because by default kwargs is an empty dict.

and

```python
kwargs['bounds']['north']
```

even if 'bounds' is set, we do not know if this is a dict to attribute ['north']

Here you can test if **kwargs** is empty and then initialize it:

```python
def add_rectangle(..., **kwargs):
    kwargs.setdefault('bounds', {})
    ....
```

The above makes sure that if 'bounds' is not in kwargs it will be set to an empty dict, if it exists the **setdefault** does nothing